pkg/controller/certificates/trigger/policies/gatherer_test.go

```diff
@@ -0,0 +1,256 @@
+			mockListRequests: mockRequestLister("default", "", []*cmapi.CertificateRequest{
+				{ObjectMeta: metav1.ObjectMeta{
+					OwnerReferences: []metav1.OwnerReference{{UID: "uid-1", Controller: pointer.BoolPtr(true)}},
+					Annotations:     map[string]string{"cert-manager.io/certificate-revision": "1"}},
+				},
+			}, nil),
+			wantCurrentRevisionRequest: &cmapi.CertificateRequest{ObjectMeta: metav1.ObjectMeta{
+				OwnerReferences: []metav1.OwnerReference{{UID: "uid-1", Controller: pointer.BoolPtr(true)}},
+				Annotations:     map[string]string{"cert-manager.io/certificate-revision": "1"}},
+			},
+			wantSecret: &v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "secret-1"}},
+		},
+		{
+			name:             "should return error when multiple certificaterequests found",
+			givenCrt:         &cmapi.Certificate{ObjectMeta: metav1.ObjectMeta{UID: "uid-1"}, Status: cmapi.CertificateStatus{Revision: ptr(1)}},
+			mockSecretLister: mockSecretLister("default", "", nil, nil),
+			mockListRequests: mockRequestLister("default", "", []*cmapi.CertificateRequest{
+				{ObjectMeta: metav1.ObjectMeta{OwnerReferences: []metav1.OwnerReference{{UID: "uid-1", Controller: pointer.BoolPtr(true)}}, Annotations: map[string]string{"cert-manager.io/certificate-revision": "1"}}},
+				{ObjectMeta: metav1.ObjectMeta{OwnerReferences: []metav1.OwnerReference{{UID: "uid-1", Controller: pointer.BoolPtr(true)}}, Annotations: map[string]string{"cert-manager.io/certificate-revision": "1"}}},
+			}, nil),
+			wantErr: "multiple CertificateRequest resources exist for the current revision, not triggering new issuance until requests have been cleaned up",
+		},
+		{
+			name:             "should return error when the list func returns an error",
+			givenCrt:         &cmapi.Certificate{ObjectMeta: metav1.ObjectMeta{UID: "uid-1"}, Status: cmapi.CertificateStatus{Revision: ptr(1)}},
+			mockSecretLister: mockSecretLister("default", "", nil, nil),
+			mockListRequests: mockRequestLister("default", "", []*cmapi.CertificateRequest{}, fmt.Errorf("some error from certificates.List that is not not_found")),
+			wantErr:          "some error from certificates.List that is not not_found",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := &Gatherer{
+				CertificateRequestLister: tt.mockListRequests(t),
+				SecretLister:             tt.mockSecretLister(t),
+			}
+
+			got, gotErr := g.DataForCertificate(context.Background(), tt.givenCrt)
+
+			if tt.wantErr != "" {
+				assert.Error(t, gotErr)
+				assert.EqualError(t, gotErr, tt.wantErr)
+				return
+			}
+
+			require.NoError(t, gotErr)
+			assert.Equal(t, tt.wantCurrentRevisionRequest, got.CurrentRevisionRequest)
+			assert.Equal(t, tt.wantSecret, got.Secret)
+			assert.Equal(t, tt.givenCrt, got.Certificate, "input cert should always be equal to returned cert")
+		})
+	}
+}
+
+// Unfortunately, pointer.IntPtr does not exist.
+func ptr(i int) *int {
```

Comment:
When we use the generators, we can make use of `gen.SetCertificateRevision` instead of this func